Automattic/wp-calypso
# Install Themes: Use feature check to determine URL

#### Changes proposed in this Pull Request

* Updates the Install Theme button to use a feature check
* Removes unused selector `isSiteOnAtomicPlan `
* Removes unused selector `siteCanUploadThemesOrPlugins`

#### Testing instructions



* Navigate to https://container-zealous-chandrasekhar.calypso.live/themes and select a site
* Look for the Install Themes button in the top right corner of the screen.
* On WoA sites it should point to wp-admin's theme installer.



Related to p4TIVU-a66-p2
Depends on https://github.com/Automattic/wp-calypso/pull/63563
